client/pages/park-activities.jsx

```diff
@@ -5,7 +5,7 @@ function Activity(props) {
 
   return (
     <div className="form-check">
-      <input className="form-check-input" type="checkbox" value={name} id={id}></input>
+      <input className="form-check-input" type="checkbox" value={name} id={id} onChange={() => props.handleCheck(event)}></input>
```

Comment:
This is going to seem like a nitpick, but it's not. The `ParkActivities` component doesn't need to know that a JavaScript event was fired, it wants to know whether or not to include an itinerary item. It's best to only pass around a useful amount of information when possible. In this case, it's much better to hide the fact an event is firing.

```jsx
function Activity(props) {
  const { name, id } = props.activity;
  const handleChange = event => {
    const { checked } = event.target; // handle the event within the Activity.
    props.onCheck({ name, checked }); // phone home with only the details that matter.
  };
  return (
    <div className="form-check">
      <input className="form-check-input" type="checkbox" value={name} id={id} onChange={handleChange}></input>
      <label className="form-check-label" htmlFor={id}>
        {name}
      </label>
    </div>
  );
}

// now you can simplify the `ParkActivities` a bit.

  handleCheck(activity) {
    const itinerary = activity.checked
      ? this.state.itinerary.concat(activity.name)
      : this.state.itinerary.filter(thingToDo => thingTodo !== activity.name);
  }
```